tsl/src/compression/create.c

```diff
@@ -245,11 +247,21 @@ compresscolinfo_init(CompressColInfo *cc, Oid srctbl_relid, List *segmentby_cols
 					 errhint("The timescaledb.compress_segmentby option must reference a valid "
 							 "column.")));
 		}
+
+		/* check if segmentby columns are distinct. */
+		if (colindex[col_attno - 1] != 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_SYNTAX_ERROR),
+					 errmsg("duplicate column name \"%s\"", NameStr(col->colname)),
+					 errhint("The timescaledb.compress_segmentby option must reference distinct "
+							 "column.")));
 		segorder_colindex[col_attno - 1] = i++;
+		colindex[col_attno - 1] = i;
```

Comment:
Shouldn't this be?
```suggestion
		segorder_colindex[col_attno - 1] = i;
		colindex[col_attno - 1] = i++;
```

It does not really make a difference since the value just has to be non-zero, but if you start to debug this, it will look strange. You could also just set it to 1 (or any other non-zero value).

Reply:
Yes I agree, changed this to `colindex[col_attno - 1] = 1`.